src/integrationtests/java/com/aws/greengrass/integrationtests/status/FleetStatusServiceSetupTest.java

```diff
@@ -0,0 +1,127 @@
+import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import com.aws.greengrass.util.Coerce;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.github.grantwest.eventually.EventuallyLambdaMatcher.eventuallyEval;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({GGExtension.class, MockitoExtension.class})
+class FleetStatusServiceSetupTest extends BaseITCase {
+    private static Kernel kernel;
+    private static DeviceConfiguration deviceConfiguration;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private AtomicReference<FleetStatusDetails> fleetStatusDetails;
+    @Mock
+    private MqttClient mqttClient;
+
+    @BeforeEach
+    void setupKernel() throws Exception {
+        fleetStatusDetails = new AtomicReference<>();
+        kernel = new Kernel();
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel,
+                FleetStatusServiceSetupTest.class.getResource("onlyMain.yaml"));
+        kernel.getContext().put(MqttClient.class, mqttClient);
+
+        when(mqttClient.publish(any(PublishRequest.class))).thenAnswer(i -> {
+            Object argument = i.getArgument(0);
+            PublishRequest publishRequest = (PublishRequest) argument;
+            try {
+                fleetStatusDetails.set(OBJECT_MAPPER.readValue(publishRequest.getPayload(), FleetStatusDetails.class));
+            } catch (JsonMappingException ignored) {
+            }
+            return CompletableFuture.completedFuture(0);
+        });
+    }
+
+    @AfterEach
+    void afterEach() {
+        kernel.shutdown();
+    }
+
+    @Test
+    void GIVEN_kernel_deployment_WHEN_device_provisioning_completes_before_kernel_launches_THEN_thing_details_uploaded_to_cloud()
+            throws Exception {
+
+        deviceConfiguration = new DeviceConfiguration(kernel, "ThingName", "xxxxxx-ats.iot.us-east-1.amazonaws.com",
+                "xxxxxx.credentials.iot.us-east-1.amazonaws.com", "privKeyFilePath", "certFilePath", "caFilePath",
+                "us-east-1", "roleAliasName");
+        kernel.getContext().put(DeviceConfiguration.class, deviceConfiguration);
+        kernel.launch();
```

Comment:
launching the kernel is slow. What can we do in these tests to combine or reduce?

Reply:
I can combine 2 of the tests. The 3rd needs different pre-launch setup. Will make the change.